src/clients/QBittorrent.ts

```diff
@@ -142,68 +138,78 @@ export default class QBittorrent implements TorrentClient {
-		if (response.status === 403 && retries > 0) {
+		let response: Response = new Response();
+		try {
+			response = await fetch(`${this.url.href}${path}`, {
+				method: "post",
+				headers: { Cookie: this.cookie, ...headers },
+				body,
+			});
+			if (response.status === 403 && retries > 0) {
+				logger.verbose({
+					label: Label.QBITTORRENT,
+					message:
+						"Received 403 from API. Logging in again and retrying",
+				});
+				await this.login();
+				return this.request(path, body, headers, retries - 1);
+			}
+		} catch (e) {
 			logger.verbose({
 				label: Label.QBITTORRENT,
-				message: "received 403 from API. Logging in again and retrying",
+				message: `Request failed: ${e.message}`,
 			});
-			await this.login();
-			return this.request(path, body, headers, retries - 1);
+			if (retries > 0) {
+				return this.request(path, body, headers, retries - 1);
+			}
 		}
 		return response.text();
 	}
 
-	async setUpCrossSeedCategory(ogCategoryName: string): Promise<string> {
-		const { linkingCategory } = getRuntimeConfig();
-		if (!ogCategoryName) return "";
-		if (
-			ogCategoryName.endsWith(TORRENT_CATEGORY_SUFFIX) ||
-			ogCategoryName === linkingCategory
-		)
-			return ogCategoryName;
-
-		const categoriesStr = await this.request("/torrents/categories", "");
-		const categories: Record<string, Category> = JSON.parse(categoriesStr);
-		const ogCategory = categories[ogCategoryName];
-		const newCategoryName = `${ogCategoryName}${TORRENT_CATEGORY_SUFFIX}`;
-		const maybeNewCategory = categories[newCategoryName];
-
-		if (!ogCategory.savePath) {
-			logOnce(`qbit/cat/no-save-path/${ogCategoryName}`, () => {
-				logger.warn(
-					`qBittorrent category "${ogCategoryName}" has no save path. Set a save path to prevent Missing Files errors.`,
-				);
-			});
+	private async getTagsForNewTorrent(
+		searchee: Searchee,
+		injectionConfiguration: TorrentConfiguration,
+	): Promise<string> {
+		const { duplicateCategories } = getRuntimeConfig();
+		const { category } = injectionConfiguration;
+
+		/* get the original category if torrent based - cuz otherwise we
+		 * use 'category'. this addresses path being truthy but still being
+		 * torrent searchee
+		 */
+		const categoryForTagging = searchee.infoHash
+			? (
+					await this.getTorrentConfiguration(
+						searchee as SearcheeWithInfoHash,
+					)
+				).category
+			: category;
+
+		if (!categoryForTagging) {
+			return TORRENT_TAG;
+		}
+		if (category.endsWith(TORRENT_CATEGORY_SUFFIX)) {
+			if (duplicateCategories) {
+				return `${TORRENT_TAG},${categoryForTagging}`;
+			} else {
+				return TORRENT_TAG;
+			}
 		}
 
-		if (maybeNewCategory?.savePath === ogCategory.savePath) {
-			// setup is already complete
-		} else if (maybeNewCategory) {
-			await this.request(
-				"/torrents/editCategory",
-				`category=${newCategoryName}&savePath=${ogCategory.savePath}`,
-				X_WWW_FORM_URLENCODED,
-			);
+		if (searchee.path) {
+			return `${TORRENT_TAG},${TORRENT_TAG}-data`;
```

Comment:
we're still using cross-seed-data?

Reply:
Yeah there requests from it to preserve all the filtering options in v5.

Reply:
What is the use case?

Reply:
Wanting to distinguish data specific filtering by tag since all torrents go to the same category now.

Reply:
but why do they want to distinguish data specific filtering? do they need to separate the torrents for some reason?

Reply:
I have no idea these qbit people are nuts.

Reply:
@bakerboy448 could you shed some light here? AFAIK, other than this, we've tried pretty hard to make hard links behave the same way whether the original files are data based or torrent based. why do people need to separate them?

Reply:
I believe it was @yammes08 and @bakerboy448 both who said that they thought this was the way to preserve ALL of the potential filtering and behaviors of v5.

Why this would be necessary to do, I don't know, but it was possible and that seemed to be a thing they wanted to do.

I don't know the use-case beyond "it was possible" lol

Reply:
This was to preserve the same granularity for filtering that exists for v5. So for example, with the existing configuration you can filter on:
- all cross-seeds (cross-seed tag)
- cross-seed data matches (cross-seed-data category)
- cross-seeds from a specific category (eg. movies.cross-seed)

Now that in in v6 all torrents, if linking is successful, will move into the same category, the additional tags provide an alternative mechanism to perform the same granularity of filters as is possible in v5.

It's not that it is something we/I specifically wanted, but there will be users out there who make use of this level of filtering. The proposed configuration retains that, whilst still reaping the benefits of linking.

Reply:
defer to yammes and davo on this as they've been closer in testing / identifying functionality than I.

Reply:
So far nobody has actually said they need this, so I'm inclined to leave it out. it'll be very easy to add back in if someone has an important use case for it.